src/main/flight/rpm_filter.c

```diff
@@ -45,162 +44,119 @@
 
 #include "rpm_filter.h"
 
-#define RPM_FILTER_MAXHARMONICS 3
-#define SECONDS_PER_MINUTE      60.0f
-#define ERPM_PER_LSB            100.0f
-#define MIN_UPDATE_T            0.001f
+#define RPM_FILTER_HARMONICS_MAX 3
+#define RPM_FILTER_DURATION_S    0.001f  // Maximum duration allowed to update all RPM notches once
+#define SECONDS_PER_MINUTE       60.0f
+#define ERPM_PER_LSB             100.0f
 
 
-static pt1Filter_t rpmFilters[MAX_SUPPORTED_MOTORS];
+typedef struct rpmFilter_s {
 
-typedef struct rpmNotchFilter_s {
+    int numHarmonics;
```

Comment:
```suggestion
    uint8_t numHarmonics;
```
OR
```suggestion
    int8_t numHarmonics;
```
Is `int` not using 16bits?

Reply:
Anyway wonder why they are defined different as in [src/main/pg/rpm_filter.h](https://github.com/betaflight/betaflight/pull/11765/files#diff-9f5dd009731aca7edab27b7bc033a23634a7b6ddc80b09d0df57edf818372114)

Reply:
I made these variables use the machine word size to speed up the processing. uint8_t and uint16_t need additional range checks, which might slow down the compiled program (veeeery slightly but still). In the PG we use uint8_t and uint16_t to save memory. But in the actual processing part we want the speed of machine word integers.